Formula/espflash.rb

```diff
@@ -0,0 +1,37 @@
+class Espflash < Formula
+  desc "Serial flasher utility for Espressif SoCs and modules based on esptool.py"
+  homepage "https://github.com/esp-rs/espflash"
+  url "https://github.com/esp-rs/espflash/archive/refs/tags/v2.0.0.tar.gz"
+  sha256 "01b65bdd7c22b422d18db6cf186cb788151f0ca0eda9533ff9478aaf7255a2cf"
+  license any_of: ["MIT", "Apache-2.0"]
+
+  depends_on "rust" => :build
+
+  on_macos do
+    depends_on "libuv" => :build
+  end
+
+  on_linux do
+    depends_on "pkg-config" => :build
+    depends_on "systemd"
+  end
+
+  resource("testdata") do
+    url "https://raw.githubusercontent.com/esp-rs/espflash/v2.0.0/espflash/tests/resources/esp32_hal_blinky"
+    sha256 "e869d05c3b4e670bb9d9f2ef304e7f32371bc4a2cff6cb90461546dfa3380473"
+  end
+
+  def install
+    system "cargo", "install", *std_cargo_args(path: "espflash")
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/espflash --version")
+    assert_match "A command-line tool for flashing Espressif devices", shell_output("#{bin}/espflash --help")
+    resource("testdata").stage do
+      assert_match "",
+        pipe_output("#{bin}/espflash flash esp32_hal_blinky --port COMX 2> error.txt")
+      assert_match "espflash::connection_failed", shell_output("cat error.txt")
```

Comment:
```suggestion
      assert_match "espflash::connection_failed",
        shell_output("#{bin}/espflash flash esp32_hal_blinky --port COMX 2>&1")
```

Reply:
`shell_output` will panic when command raise error, switch to `pipe_output` solve the problem.

Reply:
> It's wrong though. You should use `shell_output` and can pass the expected return code as a second argument to the function

oh I see, my bad, didn't read the [doc](https://rubydoc.brew.sh/Homebrew/Assertions.html#shell_output-instance_method) thoroughly